src/hal/environment.cpp

```diff
@@ -18,50 +18,40 @@ void OswHal::Environment::updateProviders() {
         if(this->tempSensor == nullptr or this->tempSensor->getTemperatureProviderPriority() < d->getTemperatureProviderPriority())
             this->tempSensor = d;
     }
-#ifndef NDEBUG
     if(!this->tempSensor)
-        Serial.println(String(__FILE__) + ": Temperature API enabled, but no provider available!");
-#endif
+        OSW_LOG_W("Temperature API enabled, but no provider available!");
 #endif
 #if OSW_PLATFORM_ENVIRONMENT_ACCELEROMETER == 1
     for(auto& d : *OswAccelerationProvider::getAllAccelerationDevices()) {
         if(this->accelSensor == nullptr or this->accelSensor->getAccelerationProviderPriority() < d->getAccelerationProviderPriority())
             this->accelSensor = d;
     }
-#ifndef NDEBUG
-    if(!this->tempSensor)
-        Serial.println(String(__FILE__) + ": Acceleration API enabled, but no provider available!");
-#endif
+    if(!this->accelSensor)
+        OSW_LOG_W("Acceleration API enabled, but no provider available!");
```

Comment:
In this case, the conditional statements are always compiled and always passed through the conditional statements, with or without debug flags. Previously, if statements were compiled only when the NDEBUG flag was declared. Can this also be included as a function or condition to output the log?
Use default param??
```
OSW_LOG_W("Acceleration API enabled, but no provider available!",!this->accelSensor);
```
like this...

Reply:
No, the statements are not compiled in builds, which have the `NDEBUG`-flag. Because in those builds, the `OSW_LOG_W` will be expanded into nothing. So practically there is written: `if(!this->accelSensor);`, which will be removed by the compiler, as the condition does not cause any changes to the object / state.